### ISSUE TITLE
Perhaps stop punning on Base functions?

### ISSUE THREAD

**Issue by KristofferC** [JuliaFEM/FEMBasis.jl#26]
FEMBasis does some "puns" on Base functions, for example:

```
        function Base.size(::Type{$name})
            return ($D, $N)
        end

        function Base.size(::Type{$name}, j::Int)
            j == 1 && return $D
            j == 2 && return $N
        end

        function Base.length(::Type{$name})
            return $N
        end
```

`length` is typically defined for collections (or at least when the instance of the type is a collection). `AbstractBasis` is not iterable and it probably shouldn't be. Instead of defining these functions I think they should have their own names `ndim`, `nbasis`.
Overloading Base functions are typically meant so you can write generic code but there is no way to write generic code with puns like this.

**Comment by KristofferC** [comment]
Another thing is that `length(...)` is usually the same as `prod(size(...))` which this breaks.

**Comment by TeroFrondelius** [comment]
Ok. This needs to change. Do you have a proposal or an idea how to change?

**Comment by ahojukka5** [comment]
Yes, we could do this somehow differently. This convention originates from the need to know proper dimensions of workspace when evaluating e.g. basis functions.
